pkg/apis/kops/cluster.go

```diff
@@ -205,6 +205,21 @@ type ClusterSpec struct {
 	RollingUpdate *RollingUpdate `json:"rollingUpdate,omitempty"`
 	// ClusterAutoscaler defines the cluster autoscaler configuration.
 	ClusterAutoscaler *ClusterAutoscalerConfig `json:"clusterAutoscaler,omitempty"`
+
+	// IAMRolesForServiceAccount defines the IRSA configuration.
+	IAMRolesForServiceAccounts *IAMRolesForServiceAccountsConfig `json:"iamRolesForServiceAccounts,omitempty"`
+}
+
+// InstanceRoleForServiceAccountConfig defines the IRSA configuration.
+type IAMRolesForServiceAccountsConfig struct {
+	ServiceAccounts []ServiceAccountMapping `json:"serviceAccounts,omitempty"`
```

Comment:
I think this should go into `IAMSpec` instead of its own struct

Reply:
We went a bit back and fort on this. I am not against moving it. But the intention is to merge this with the fields from #11194 so has to consider then if it should be still a substruct or not.

Reply:
I commented in #11194 that the `OIDCProviderLocation` field is not specific to IRSA. It should not be merged into any IAM-specific substruct.